project-base/storefront/helpers/seo/generateCanonicalUrl.ts

```diff
@@ -1,32 +1,47 @@
-import {
-    getQueryWithoutSlugTypeParameterFromParsedUrlQuery,
-    getUrlWithoutGetParameters,
-} from 'helpers/parsing/urlParsing';
+import { getUrlWithoutGetParameters } from 'helpers/parsing/urlParsing';
 import { getStringWithoutTrailingSlash } from 'helpers/parsing/stringWIthoutSlash';
-import { INTERNAL_QUERY_PARAMETERS } from 'helpers/queryParamNames';
 import { NextRouter } from 'next/router';
+import {
+    PAGE_QUERY_PARAMETER_NAME,
+    SEARCH_QUERY_PARAMETER_NAME,
+    FILTER_QUERY_PARAMETER_NAME,
+    SORT_QUERY_PARAMETER_NAME,
+} from 'helpers/queryParamNames';
+
+const DEFAULT_CANONICAL_QUERY_PARAMS = [
+    PAGE_QUERY_PARAMETER_NAME,
+    SEARCH_QUERY_PARAMETER_NAME,
+    FILTER_QUERY_PARAMETER_NAME,
+    SORT_QUERY_PARAMETER_NAME,
+] as const;
+
+export type CanonicalQueryParameters = (typeof DEFAULT_CANONICAL_QUERY_PARAMS)[number][];
 
-export const generateCanonicalUrl = (router: NextRouter, url: string): string | null => {
+export const generateCanonicalUrl = (
+    router: NextRouter,
+    url: string,
+    canonicalQueryParams?: CanonicalQueryParameters,
+): string | null => {
     const newQueryOverwrite: Record<string, string> = {};
-    const queryWithoutAllParameter = getQueryWithoutSlugTypeParameterFromParsedUrlQuery(router.query);
+    const queries = router.query;
```

Comment:
Do we not care about the removal of the slug type parameter?

Reply:
Nope, that's the point. We are filtering this out anyway.